draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,76 +60,291 @@ collector). Thus it is up to each deployment of DAP to securely implement its
+    can be found. Defined in I-D.draft-ietf-ppm-dap-02. */
+    Url aggregator_endpoints<1..2^16-1>;
+
+    /* This determines the query type for batch selection and the
+    properties that all batches for this task must have. Defined in
+    I-D.draft-ietf-ppm-dap-02. */
+    QueryConfig query_config;
+
+    /* The maximum number of times a batch of reports may be queried by
+    the Collector. */
+    uint16 max_batch_lifetime;
+
+    /* Time up to which Clients are allowed to upload to this task. See
+    https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/pull/304. Defined
+    in I-D.draft-ietf-ppm-dap-02. */
+    Time task_expiration;
+
+    /* A codepoint defined in I-D.draft-irtf-cfrg-vdaf-03 or reserved
+    for private use. */
+    VdafType vdaf_type;
+
+    /* Additional parameters relevant for the vdaf_type. */
+    opaque vdaf_config<1..2^16-1>;
+} TaskConfig;
+
+struct {
+    QueryType query_type;    // Defined in I-D.draft-ietf-ppm-dap-02
+    Duration time_precision; // Defined in I-D.draft-ietf-ppm-dap-02
+    uint32 min_batch_size;
+    select (query_type) {
+        case time-interval: Empty;
+        case fixed-size:     uint32 max_batch_size;
+    }
+} QueryConfig;
 ~~~
 
+The purpose of `TaskConfig` is to define all parameters that are necessary
+for configuring an aggregator. It includes all the fields to be
+associated with a task (see task configuration in
+{{?DAP=I-D.draft-ietf-ppm-dap-01}}.). Besides, `TaskConfig` also includes fields
+that useful for configuring a task in-band:
+
+* An opaque `task_info` that is specific to a task. For e.g. this can be a
+  string describing the purpose of this task.
+
+* An opaque `vdaf_config` that contains any VDAF specific parameters for the
+  chosen `vdaf_type`.
+
+The codepoints for standardized (V)DAFs are listed below:
+
+~~~
+/* Codepoint for each standardized VDAF. Defined in
+ I-D.draft-irtf-cfrg-vdaf-03. */
+enum {
+    prio3-aes128-count(0x00000000),
+    prio3-aes128-sum(0x00000001),
+    prio3-aes128-histogram(0x00000002),
+    poplar1-aes128(0x00001000),
+    (2^32-1)
+} VdafType;
+~~~
+
+The structure of the `vdaf_config` field is not specified in this document,
+instead it needs to be defined by the VDAF itself. For VDAFs specified
+in {{?VDAF=I-D.draft-irtf-cfrg-vdaf-01}}, implementations SHOULD use the
+following structure:
+
+~~~
+struct {
+    select (vdaf_type) {
+        case prio3-aes128-count: Empty;
+        case prio3-aes128-sum: uint8 bits;
+        case prio3-aes128-histogram: uint64 buckets<8, 2^24-8>;
+        case poplar1-aes128: uint16 bits;
+    }
+} VdafConfig;
+~~~
+
+> OPEN ISSUE: Should DP parameters be defined as a different "dimension" to
+> VDAF, given that it's likely various DP mechanisms can be applied to any VDAF.
+> See issue [#94](https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/94) for
+> discussion.
+
+The definition of Time, Duration, Url, QueryType follow those in
+{{?DAP=I-D.draft-ietf-ppm-dap-01}}.
```

Comment:
```suggestion
{{?DAP=I-D.draft-ietf-ppm-dap-02}}.
```

Reply:
This draft isn't out yet, so we'll have to update this later :)

Reply:
Fair enough, I'm just not clear because there is a mixture of reference to DAP-01 and DAP-02.

Reply:
TOTALLY understandable! We were hoping to cut DAP-02 last week, but it looks like we'll merge this PR before then.